### DISCUSSION TITLE
提交bug。docker-compose不能使用

### DISCUSSION THREAD

**Discussion by alilovetaozi** [fofapro/vulfocus#319] [Announcements]
https://github.com/fofapro/vulfocus/blob/00e09e65784d5bf2217a9e458276f0de81354999/vulfocus-api/layout_image/views.py#L981 中original_yml=json.dumps(yaml.load(args,Loader=yaml.FullLoader)) original_yml 是 docker-compose转成对象后又转成json的字符串。984行调用tasks.create_compose_task。
![图片](https://user-images.githubusercontent.com/42528382/203559811-5d1beeff-cecb-410b-bee0-eabfcc67d38d.png)
https://github.com/fofapro/vulfocus/blob/master/vulfocus-api/tasks/tasks.py#L140运行到这里，按照原文的意思，这里 args_yaml应该是一个python对象，args_yaml = new_yaml.load(json.loads(image_info.original_yml))处理时。先将前面的json字符串格式转成json对象，然后用yaml.load（）加载，这里面并不是yaml格式字符串。看后面的处理，这里直接用json对象也行。于是删除yaml.load 改成   args_yaml = json.loads(image_info.original_yml) 就行了。
![图片](https://user-images.githubusercontent.com/42528382/203560716-84456b42-5087-4204-adec-5b807e3ddfcd.png)
修改后这里就成功了，并且也能用
![图片](https://user-images.githubusercontent.com/42528382/203561763-e77fd50c-2ae8-4c22-990b-3f1e1f77c231.png)

![图片](https://user-images.githubusercontent.com/42528382/203562171-52e147af-28c1-4f36-ba81-2b07b126ef65.png)
![图片](https://user-images.githubusercontent.com/42528382/203562235-ab2b0dd1-0d11-4f7a-9f7b-95bb161942a9.png)
![图片](https://user-images.githubusercontent.com/42528382/203562453-da7f5fac-7c8a-454e-91c8-5d9048121e5c.png)

**Comment by xxxxbxxxxx** [comment]
`docker-compose` 不推荐使用，建议制作镜像的时候采用 `Docker in Docker`。

**Comment by alilovetaozi** [comment]
有这功能就想着用一下。其实用起来还行，可能还有bug。有时间再调一下。
